tests/test_circulationapi.py

```diff
@@ -75,6 +76,10 @@ def test_borrow_sends_analytics_event(self):
             }
         }
         with temp_config(config) as config:
+            provider = MockAnalyticsProvider()
+            analytics = Analytics.initialize(
+                ['core.mock_analytics_provider'], config
+            )
```

Comment:
The `Analytics` object is a singleton, so right now any test that wants to use a specific analytics provider needs to call its `initialize` method (not its constructor, as I saw in some unused code I removed).

What we really need is a `with temp_analytics_provider(analytics_class)` construct that handles the setup and teardown, but this is good enough for now.